diff -urN crosstool-ng-1.17.0~orig/scripts/functions crosstool-ng-1.17.0/scripts/functions
--- crosstool-ng-1.17.0~orig/scripts/functions	2013-01-07 13:44:29.370933224 +0100
+++ crosstool-ng-1.17.0/scripts/functions	2013-01-07 21:54:30.044459907 +0100
@@ -927,6 +927,7 @@
         .tar.gz|.tgz) gzip -dc "${full_file}" | CT_DoExecLog FILE tar "${tar_opts[@]}" -f -;;
         .tar)         CT_DoExecLog FILE tar "${tar_opts[@]}" -f "${full_file}";;
         /.git)        CT_ExtractGit "${basename}" "${@}";;
+        "")           CT_DoExecLog FILE cp -a ${full_file}/. ${basename}/ ;;
         *)            CT_DoLog WARN "Don't know how to handle '${basename}${ext}': unknown extension"
                       return 1
                       ;;
