--- crosstool-ng-1.17.0/scripts/build/binutils/elf2flt.sh~orig	2012-12-08 14:19:00.016217802 +0100
+++ crosstool-ng-1.17.0/scripts/build/binutils/elf2flt.sh	2012-12-08 14:19:13.792217455 +0100
@@ -49,7 +49,7 @@
 
     elf2flt_opts+=( "host=${CT_BUILD}" )
     elf2flt_opts+=( "prefix=${CT_BUILDTOOLS_PREFIX_DIR}" )
-    elf2flt_opts+=( "binutils_bld=${CT_BUILD_DIR}/build-binutils-build-${CT_HOST}" )
+    elf2flt_opts+=( "binutils_bld=${CT_BUILD_DIR}/build-binutils-build-${CT_BUILD}" )
 
     do_elf2flt_backend "${elf2flt_opts[@]}"
 
