diff -r 9303953d3983 ct-ng.in
--- a/ct-ng.in	Wed Sep 26 16:41:02 2012 +0200
+++ b/ct-ng.in	Thu Sep 27 13:40:27 2012 +0200
@@ -16,8 +16,9 @@
 
 # Paths and values set by ./configure
 # Don't bother to change it other than with a new ./configure!
-export CT_LIB_DIR:=@@CT_LIBDIR@@
-export CT_DOC_DIR:=@@CT_DOCDIR@@
+ct_prefix=$(shell dirname $(MAKEFILE_LIST))/..
+export CT_LIB_DIR:=$(ct_prefix)@@CT_LIBDIR@@
+export CT_DOC_DIR:=$(ct_prefix)@@CT_DOCDIR@@
 
 # This is crosstool-NG version string
 export CT_VERSION:=@@CT_VERSION@@
