--- crosstool-ng-1.17.0/scripts/build/libc/uClibc.sh~orig	2013-04-12 12:48:08.218978664 +0200
+++ crosstool-ng-1.17.0/scripts/build/libc/uClibc.sh	2013-04-12 13:03:06.394956155 +0200
@@ -373,6 +373,31 @@
             ;;
     esac
 
+    # As we are building libc with cc-core-pass-2, which does not seem to
+    # enforce proper arch selection, at least not on m68k (ColdFire), we pass
+    # in any additional compiler flags needed.  We assume that float and abi
+    # flags are handled properly without needing to fix them up here.
+    if [ -n "${CT_ARCH_ARCH_CFLAG}" ] ; then
+        cat <<-ENDSED
+		s/^(UCLIBC_EXTRA_CFLAGS=".*)"$/\\1 ${CT_ARCH_ARCH_CFLAG}"/
+		ENDSED
+    fi
+    if [ -n "${CT_ARCH_CPU_CFLAG}" ] ; then
+        cat <<-ENDSED
+		s/^(UCLIBC_EXTRA_CFLAGS=".*)"$/\\1 ${CT_ARCH_CPU_CFLAG}"/
+		ENDSED
+    fi
+    if [ -n "${CT_ARCH_TUNE_CFLAG}" ] ; then
+        cat <<-ENDSED
+		s/^(UCLIBC_EXTRA_CFLAGS=".*)"$/\\1 ${CT_ARCH_TUNE_CFLAG}"/
+		ENDSED
+    fi
+    if [ -n "${CT_ARCH_FPU_CFLAG}" ] ; then
+        cat <<-ENDSED
+		s/^(UCLIBC_EXTRA_CFLAGS=".*)"$/\\1 ${CT_ARCH_FPU_CFLAG}"/
+		ENDSED
+    fi
+
     # We always want ctor/dtor
     cat <<-ENDSED
 		s/^# (UCLIBC_CTOR_DTOR) is not set/\\1=y/
