diff -urN libtool-2.2.10/configure.ac libtool-2.2.10.mod/configure.ac
--- libtool-2.2.10/configure.ac	2010-06-09 15:00:04.000000000 +0200
+++ libtool-2.2.10.mod/configure.ac	2012-10-10 01:35:44.000000000 +0200
@@ -122,7 +122,7 @@
 
 dnl These are bootstrap requirements! Once built, libtool may work with
 dnl much older releases of autoconf and automake.  See release notes.
-AM_INIT_AUTOMAKE([1.10.1 gnu subdir-objects dist-lzma])
+AM_INIT_AUTOMAKE([1.10.1 gnu subdir-objects])
 
 dnl We use m4sh to generate libtool's portable shell scripts
 AC_SUBST([M4SH], ['$(AUTOM4TE) --language=m4sh'])
