Index: crosstool-ng-1.12.4/scripts/build/arch/x86.sh
===================================================================
--- crosstool-ng-1.12.4.orig/scripts/build/arch/x86.sh	2011-11-25 13:47:23.346456701 +0100
+++ crosstool-ng-1.12.4/scripts/build/arch/x86.sh	2011-11-25 13:48:07.848755594 +0100
@@ -19,4 +19,17 @@
             *)                            CT_TARGET_ARCH=i586;;
         esac
     fi
+
+    # with-float is not possible for x86
+    case "${CT_ARCH_FLOAT}" in
+        hard)
+            CT_ARCH_FLOAT_CFLAG=
+            CT_ARCH_WITH_FLOAT=
+            ;;
+        soft)
+            CT_ARCH_FLOAT_CFLAG="-msoft-float"
+            CT_ARCH_WITH_FLOAT=
+            ;;
+    esac
+
 }
