Index: crosstool-ng-1.12.3/config/cc/gcc.in
===================================================================
--- crosstool-ng-1.12.3.orig/config/cc/gcc.in	2011-10-13 11:02:39.502902075 +0200
+++ crosstool-ng-1.12.3/config/cc/gcc.in	2011-10-13 11:03:44.612997561 +0200
@@ -36,9 +36,9 @@
 # Don't remove next line
 # CT_INSERT_VERSION_BELOW
 
-config CC_V_linaro_4_6_2011_07
+config CC_V_linaro_4_6_2011_10
     bool
-    prompt "linaro-4.6-2011.07 (EXPERIMENTAL)"
+    prompt "linaro-4.6-2011.10 (EXPERIMENTAL)"
     depends on CC_GCC_SHOW_LINARO
     select CC_GCC_4_6
 
@@ -54,9 +54,9 @@
     depends on EXPERIMENTAL
     select CC_GCC_4_6
 
-config CC_V_linaro_4_5_2011_07
+config CC_V_linaro_4_5_2011_10
     bool
-    prompt "linaro-4.5-2011.07 (EXPERIMENTAL)"
+    prompt "linaro-4.5-2011.10 (EXPERIMENTAL)"
     depends on CC_GCC_SHOW_LINARO
     select CC_GCC_4_5
 
@@ -319,10 +319,10 @@
     string
 # Don't remove next line
 # CT_INSERT_VERSION_STRING_BELOW
-    default "linaro-4.6-2011.07" if CC_V_linaro_4_6_2011_07
+    default "linaro-4.6-2011.10" if CC_V_linaro_4_6_2011_10
     default "4.6.1" if CC_V_4_6_1
     default "4.6.0" if CC_V_4_6_0
-    default "linaro-4.5-2011.07" if CC_V_linaro_4_5_2011_07
+    default "linaro-4.5-2011.10" if CC_V_linaro_4_5_2011_10
     default "4.5.3" if CC_V_4_5_3
     default "4.5.2" if CC_V_4_5_2
     default "4.5.1" if CC_V_4_5_1
