diff -urN chrpath-0.14~orig/configure chrpath-0.14/configure
--- chrpath-0.14~orig/configure	2011-09-26 20:21:41.000000000 +0200
+++ chrpath-0.14/configure	2013-05-15 11:52:38.791123727 +0200
@@ -551,7 +551,7 @@
 
 # Identity of this package.
 PACKAGE_NAME='"chrpath"'
-PACKAGE_TARNAME='-chrpath-'
+PACKAGE_TARNAME='chrpath-0.14'
 PACKAGE_VERSION='"0.14"'
 PACKAGE_STRING='"chrpath" "0.14"'
 PACKAGE_BUGREPORT='pere@hungry.com'
diff -urN chrpath-0.14~orig/Makefile.am chrpath-0.14/Makefile.am
--- chrpath-0.14~orig/Makefile.am	2011-09-27 11:06:43.000000000 +0200
+++ chrpath-0.14/Makefile.am	2013-05-15 11:51:17.011125779 +0200
@@ -1,7 +1,5 @@
 SUBDIRS = testsuite deb
 
-docdir = $(prefix)/doc/$(PACKAGE)-$(VERSION)
-
 doc_DATA = AUTHORS COPYING ChangeLog INSTALL NEWS README
 
 bin_PROGRAMS = chrpath
diff -urN chrpath-0.14~orig/Makefile.in chrpath-0.14/Makefile.in
--- chrpath-0.14~orig/Makefile.in	2011-09-27 11:07:45.000000000 +0200
+++ chrpath-0.14/Makefile.in	2013-05-15 11:51:17.011125779 +0200
@@ -211,7 +211,7 @@
 builddir = @builddir@
 datadir = @datadir@
 datarootdir = @datarootdir@
-docdir = $(prefix)/doc/$(PACKAGE)-$(VERSION)
+docdir = @docdir@
 dvidir = @dvidir@
 exec_prefix = @exec_prefix@
 host = @host@
