diff -uprN crosstool-ng-1.7.1.org/scripts/build/companion_libs/cloog.sh crosstool-ng-1.7.1/scripts/build/companion_libs/cloog.sh
--- crosstool-ng-1.7.1.org/scripts/build/companion_libs/cloog.sh	2010-06-27 18:38:14.000000000 +0200
+++ crosstool-ng-1.7.1/scripts/build/companion_libs/cloog.sh	2010-07-27 21:14:20.000000000 +0200
@@ -56,6 +56,7 @@ do_cloog() {
         cloog_opts+=( --enable-shared --disable-static )
     else
         cloog_opts+=( --disable-shared --enable-static )
+        cloog_opts+=( --with-host-libstdcxx='-lstdc++' )
         cloog_LDFLAGS='-lstdc++'
     fi
 
