diff -urN crosstool-ng-1.8.0-orig/scripts/build/cc/gcc.sh crosstool-ng-1.8.0/scripts/build/cc/gcc.sh
--- crosstool-ng-1.8.0-orig/scripts/build/cc/gcc.sh	2010-08-10 12:35:03.000000000 +0200
+++ crosstool-ng-1.8.0/scripts/build/cc/gcc.sh	2010-08-10 12:41:43.000000000 +0200
@@ -360,6 +360,7 @@
         # is needed. Shoe-horn it in our LDFLAGS
         # Ditto libm on some Fedora boxen
         final_LDFLAGS='-lstdc++ -lm'
+        extra_config+=("--with-host-libstdcxx='-lstdc++ -lm")
     fi
     if [ "${CT_CC_GCC_USE_GMP_MPFR}" = "y" ]; then
         extra_config+=("--with-gmp=${CT_COMPLIBS_DIR}")
